keps/sig-node/20190226-pod-overhead.md

```diff
@@ -0,0 +1,310 @@
+- Two versions passed since introducing the functionality which deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include [conformance tests].**
+
+[conformance tests]: https://github.com/kubernetes/community/blob/master/contributors/devel/conformance-tests.md
+
+### Upgrade / Downgrade Strategy
+
+If applicable, how will the component be upgraded and downgraded? Make sure this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to make use of the enhancement?
+
+### Version Skew Strategy
+
+Set the overhead to the max of the two version until the rollout is complete.  This may be more problematic
+if a new version increases (rather than decreases) the required resources.
+
+## Implementation History
+
+Major milestones in the life cycle of a KEP should be tracked in `Implementation History`.
+Major milestones might include
+
+- the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+- the `Proposal` section being merged signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+
+## Drawbacks [optional]
+
+This KEP introduceds further complexity, and adds a field the PodSpec which users aren't expected to modify.
+
+## Alternatives [optional]
+
+In order to achieve proper handling of sandbox runtimes, the scheduler/resourceQuota handling needs to take
+into account the overheads associated with running a particular runtimeClass.
+
+### Introduce pod level resource requirements
+
+Rather than just introduce overhead, add support for general pod-level resource requirements. Pod level
+resource requirements are useful for shared resources (hugepages, memory when doing emptyDir volumes).
+
+Even if this were to be introduced, there is a benefit in keeping the overhead separate.
+ - post-pod creation handling of pod events: if runtimeClass definition is removed after a pod is created,
+  it will be very complicated to calculate which part of the pod resource requirements were associated with
+  the workloads versus the sandbox overhead.
+ - a kubernetes service provider can subsidize the charge-back model potentially and eat the cost of the
+ runtime choice, but charge the user for the cpu/memory consumed independent of runtime choice.
+
+
+### Leaving the PodSpec unchaged
```

Comment:
I had been assuming that we would merge the overhead into the podspec, but now I'm second-guessing that...